osu.Game.Rulesets.Osu/Objects/Drawables/SliderInputManager.cs

```diff
@@ -0,0 +1,254 @@
+
+        private Vector2? screenSpaceMousePosition;
+        private readonly DrawableSlider slider;
+
+        public SliderInputManager(DrawableSlider slider)
+        {
+            this.slider = slider;
+        }
+
+        /// <summary>
+        /// This component handles all input of the slider, so it should receive input no matter the position.
+        /// </summary>
+        public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => true;
+
+        protected override bool OnMouseMove(MouseMoveEvent e)
+        {
+            screenSpaceMousePosition = e.ScreenSpaceMousePosition;
+            return base.OnMouseMove(e);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+            updateTracking(isMouseInFollowArea(Tracking));
+        }
+
+        public void PostProcessHeadJudgement(DrawableSliderHead head)
+        {
+            if (!head.Judged || !head.Result.IsHit)
+                return;
+
+            if (!isMouseInFollowArea(true))
+                return;
+
+            Debug.Assert(screenSpaceMousePosition != null);
+
+            Vector2 mousePositionInSlider = slider.ToLocalSpace(screenSpaceMousePosition.Value) - slider.OriginPosition;
+
+            // When the head is hit and the mouse is in the expanded follow area, force a hit on every nested hitobject
+            // from the start of the slider that is within the follow area.
+
+            bool forceMiss = false;
+
+            foreach (var nested in slider.NestedHitObjects.OfType<DrawableOsuHitObject>())
+            {
+                // Skip nested objects that are already judged.
+                if (nested.Judged)
+                    continue;
+
+                // Stop the process when a nested object is reached that can't be hit before the current time.
+                if (nested.HitObject.StartTime > Time.Current)
+                    break;
+
+                float radius = getFollowRadius(true);
+                double objectProgress = Math.Clamp((nested.HitObject.StartTime - slider.HitObject.StartTime) / slider.HitObject.Duration, 0, 1);
+                Vector2 objectPosition = slider.HitObject.CurvePositionAt(objectProgress);
+
+                // When the first nested object that is further outside the follow area is reached,
+                // forcefully miss all other nested objects that would otherwise be valid to be hit.
+                // This covers a case of a slider overlapping itself that requires tracking to a tick on an outer edge.
+                if (forceMiss || (objectPosition - mousePositionInSlider).LengthSquared > radius * radius)
+                {
+                    nested.MissForcefully();
+                    forceMiss = true;
+                }
+                else
+                    nested.HitForcefully();
+            }
+
+            // Enable tracking, since the mouse is within the follow area (if it were expanded).
+            updateTracking(true);
```

Comment:
I just realized that it might make more sense to not start tracking if `forceMiss` is true and the cursor is not on the ball.
So it should be:
```cs
updateTracking(!forceMiss || isMouseInFollowArea(false));
```
Which is actually equivalent to
```cs
updateTracking(isMouseInFollowArea(!forceMiss));
```

Reply:
@smoogipoo this seems valid, please check whether you agree.

Reply:
Is there a specific reason why this makes more sense to you?

When it comes to these extreme edge cases, I tend to prefer to follow the easier-to-explain mechanic. In this case - that if you're within the tracking radius at the point where the head is hit, you gain tracking.

Reply:
I feel like we are simulating the movement of the ball as if the ball was instantly reproducing the movement from the start to where it should be at the time. This means we start tracking on the head hit, and as soon as the ball is further than follow-circle away, the ball deactivates (and thus we miss all the other ticks). If the ball finishes it's movement under the cursor it should reactivate though (which is why we have `isMouseInFollowArea(false)` in that case)

Reply:
I applied this change in the recent commits.